app/Http/Controllers/TalksController.php

```diff
@@ -151,61 +153,51 @@ public function destroy($id)
 
     public function archiveIndex()
     {
-        $talks =  Auth::user()->talks()->archived()->get();
-        list($talks, $sorting_talk) = $this->sortTalks($talks);
+        $talks = $this->sortTalks(
+            Auth::user()->talks()->archived()->get()
+        );
+
+        $sorting_style = $this->sorting_style;
 
         return View::make('talks.archive')
           ->with('talks', $talks)
-          ->with('sorting_talk', $sorting_talk);
+          ->with('sorting_style', $sorting_style);
     }
 
     public function archive($id)
     {
-        $talk = Auth::user()->talks()->findOrFail($id);
-        $talk->is_archived = true;
-        $talk->save();
+        $talk = Auth::user()->talks()->findOrFail($id)->archive();
```

Comment:
Don't even need to save it to `$talk` variable, because you're not going to use it anywhere. :) So fresh. So clean.